### ISSUE TITLE
Default username in Active Directory environments has changed with 2.5.0

### ISSUE THREAD

**Issue by saxonww** [git-for-windows/git#292]
With the latest msysgit release (1.9.5), a Windows username passed to ssh.exe was simply 'username'. On 2.5.0 in an AD environment, the username is now 'DOMAIN+username'. Any ssh remote which does not explicitly define the username will probably stop working after upgrade and it won't be obvious why.

I've worked around this by using an ssh config file.

**Comment by dscho** [comment]
Hmm. I wonder which version of SSH introduced that. @saxonww where did you see the information that you need to use the `DOMAIN+` prefix now?

**Comment by saxonww** [comment]
I didn't see it anywhere in the release notes. I updated one of our build machines to 2.5.0 and saw that it couldn't pull; an `ssh -v` showed it was trying `DOMAIN+username`, which I know has never been correct in our environment.

We use 1.9.5.msysgit.1 everywhere else, which ships with OpenSSH 6.6.1.p1.

**Comment by dscho** [comment]
Oh, so `DOMAIN+username` is what it tries now, not what is correct. I understand now... I guess there must be a config option for OpenSSH to turn that behavior off.

**Comment by saxonww** [comment]
I think this is actually happening because of the move from msys -> msys2, or mingw64 or whatever the new POSIX layer is. Looking [here](https://cygwin.com/cygwin-ug-net/ntsec.html) in the 'Mapping Windows SIDs to POSIX uid/gid values' section, it is clear that this is the normal behavior (for Cygwin at least) *if the user is from a separate domain*.

I wasn't thinking when I filed this issue; this particular machine is on a different AD domain than the user I was logged in as. I've verified that if I log in as a domain-local user that username is formatted as expected. 

The old msys with the 1.x msysgit probably uses an older/different username mapping scheme that doesn't deal with domains.

Basically I am not sure if this can be fixed. Unless you think otherwise, then this can just be closed.
